app/javascript/lib/sentry-logging.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * @fileoverview Functions for logging JS errors to Sentry
+ *
+ */
+import * as Sentry from '@sentry/react'
+import getSCPContext from '~/providers/SCPContextProvider'
+
+/**
+ * Log an exception to Sentry for bad response JS fetch executions
+ * i.e. executions that results in a 404 response
+ *
+ * @param {Object} response - the response object from a failed JS fetch call
+ * @param {String} titleInfo - extra info for the title of the Sentry event
+ */
+export function logJSFetchExceptionToSentry(response, titleInfo = '') {
+  // add details from the response to the 'response info' object that will be logged in Sentry
+  Sentry.setContext('response info', {
+    status: response.status,
+    statusText: response.statusText,
+    url: response.url
+  })
+
+  shouldLog() && Sentry.captureException(new Error(`${response.status}: ${titleInfo}`))
+}
+
+/**
+ * Log an exception to Sentry for failed JS fetch executions
+ * i.e. fetches that fail and error
+ *
+ * @param {Object} response - the response object from a failed JS fetch call
+ * @param {String} titleInfo - extra info for the title of the Sentry event
+ */
+export function logJSFetchErrorToSentry(error, titleInfo = '') {
+  shouldLog() && Sentry.captureException(new Error(`${error}: ${titleInfo}`))
+}
+
+/**
+ * Determine if logging should occur based on environment
+ */
+function shouldLog() {
+  const env = getSCPContext().environment
+
+  // do not log for development or test environments
+  // to test locally comment out lines 45-47
+  if (['development', 'test'].includes(env)) {
```

Comment:
I understand that you did this to make it easier to test, this is a little cleaner:
```suggestion
  return ['development', 'test'].includes(env)
```
For testing, you could simply write `return true` above that line to skip the `env` check.

On a related note, this would mean that we're logging errors from `staging`, which we may or may not want to do.  Typically, we only use services like Sentry in production.  If we want to keep that behavior, you can rewrite line 45 as:
```
return env === 'production'
```

Reply:
I can definitely make that change if that would be better - I was following what I had seen `ErrorTracker.rb` where it has the logic:

>     # only report to Sentry if configured and not in dev or test environment
>     context = format_extra_context(*context_objects)
>     if %w(development test).include?(Rails.env) || ENV['SENTRY_DSN'].nil?
>       Rails.logger.error "Suppressing error reporting to Sentry: #{exception.class.name}:#{exception.message}, context: #{context}"
>     else
>       Raven.capture_exception(exception, user: {identifer: extract_user_identifier(user)}, extra: context)
>     end

so figured I should match that but let me know if you think I should update this to only do Prod

Reply:
No, that actually makes sense - let's leave it.